### ISSUE TITLE
Search API: truncate/remove source content from sections

### ISSUE THREAD

**Issue by stsewd** [readthedocs/readthedocs.org#10321]
## What's the problem this feature will solve?

When searching, if a result is part of a larger section, we will return the whole content of that section. For some docs, this can be big, for example this https://postofficeinquiry.dracos.co.uk/_/api/v2/search/?q=test&project=post-office-horizon-inquiry-hearings&version=latest&language=en will result in a ~2MB response (~6MB uncompressed!!).

## Describe the solution you'd like

Excluding the source for sections may not always work, since if there is a match in the title, and not in the content, we won't have a highlight result, so we won't be able to put context in the result. But maybe this is okay? Or maybe there is a way to do this in a smart way with ES?

We could also check if we could limit the amount of text returned for each section.

**Comment by humitos** [comment]
I don't think this issue is gonna affect too many projects; so I wouldn't spend too much time in the perfect solution. However, I'd say we should focus on protecting ourselves as the first step and either,

* tell ES to cancel queries with big results; and return a simple error to the user
* return a chunked response with <1Mb data to the user via the API

**Comment by humitos** [comment]
Another good first step here would be adding logging for these cases to be able to evaluate how often it happens and make a decision with that data.
